subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/m5/ToolingApiIdeaModelCrossVersionSpec.groovy

```diff
@@ -215,7 +215,16 @@ project(':impl') {
         lib.scope.scope == 'TEST'
 
         IdeaModuleDependency mod = libs.find {it instanceof IdeaModuleDependency}
-        mod.dependencyModule == project.modules.find { it.name == 'api'}
+
+        def currentVersion = GradleVersion.current()
+        if (currentVersion >= GradleVersion.version("3.1")) {
+            mod.targetModuleName == project.modules.find { it.name == 'api'}.getName()
+        }
+
+        if (currentVersion < GradleVersion.version("3.99")) {
+            mod.dependencyModule == project.modules.find { it.name == 'api'}
+        }
+
```

Comment:
@adammurdoch Please let me know if this is the direction you want me to take on all of the tests.  I will extract currentVersion into the test fixture along side targetVersion.  Note that I used 3.99 instead of 4.0 because the current version 4.0-SNAPSHOT-1 is technically before 4.0.  I'm open to suggestions if there is a better version to use to represent anything below any 4.0.  Perhaps extracting the major version only?